### DISCUSSION TITLE
Improve Pane manipulation

### DISCUSSION THREAD

**Discussion by Adriani277** [zed-industries/zed#10717] [Feature Requests]
Hello, coming from VS-code I wanted to be able to perform the following

Given I have 2 tabs open in a single Pane, I want to move the active tab into a new Pane. This is equivalent of move-pane-right
The current way to achieve this is via:
```
        "workspace::SendKeystrokes",
        "cmd-k right cmd-k cmd-left cmd-w cmd-k cmd-right
```
Which will do
1. Split Pane right
2. Move to left Pane
3. close active tab
4. move to right Pane

While this works, it seems like there should be a built-in command for this(maybe there is and I have just missed it).

Also, I have not been able to figure out how to undo the split. So have 2 Panes, I would like to merge my active pane back into the inactive one

**Comment by fmarkulin** [comment]
I highly miss this from vscode and have not switched yet because of it (and other things I commented on)

**Comment by saveman71** [comment]
additionally, there should be a way to move a existing pane left and right (up and down), but as the pane doesn't exist in the destination, there is now way to script this

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
